File: 2/solution.rb
fib = [1, 1]

def populateFib fib
	next_term = fib[-1] + fib[-2]
	unless next_term > 4_000_000
		fib.push next_term
		populateFib fib
	end
end

populateFib fib

total = 0
fib.each do |i|
	if i % 2 == 0
		total += i
	end
end

puts total

File: 2/solution.js
var fib = [1, 1];

(function populateFib() {
	var next_term = fib[fib.length - 1] + fib[fib.length - 2];
	if (next_term > 4000000) { 
		return;
	} else {
		fib.push(next_term);
		populateFib();
	}
})();

var total = 0
for (i = 0; i < fib.length; i += 1) {
	if (fib[i] % 2 === 0) {
		total += fib[i];
	}
}

console.log(total);